packages/@lwc/compiler/src/options.ts

```diff
@@ -65,7 +65,10 @@ export interface TransformOptions {
     name?: string;
     namespace?: string;
     stylesheetConfig?: StylesheetConfig;
-    experimentalDynamicComponent?: DynamicComponentConfig;
+    dynamicImportConfig?: DynamicImportConfig;
+    // TODO [#3331]: remove usage of lwc:dynamic in 246
+    experimentalDynamicDirective?: boolean;
```

Comment:
I debated about this initially but I think you're right.  Changing these compiler options will lead to unnecessary overhead in managing the release of this feature to downstream dependencies.

Reply:
I've changed `dynamicImportConfig` back to `experimentalDynamicComponents` to prevent breaking downstream consumers and to ease the enablement of this feature.

We're planning to remove the `lwc:dynamic` directive entirely in 246 and I plan to deprecate `experimentalDynamicComponent` in favor of `dynamicImportConfig` as part of it.

Reply:
You can remove this from the PR description!
<img width="901" alt="Screenshot 2023-02-10 at 12 27 57 PM" src="https://user-images.githubusercontent.com/931465/218191219-421f8842-03b5-4004-b048-81845febb187.png">